src/plugins/codecZip.nim

```diff
@@ -177,6 +185,41 @@ proc zipScan*(self: Plugin, loc: string): Option[ChalkObj] {.cdecl.} =
 
     return some(chalk)
 
+proc insertChalkBinaryIntoZip*(chalk: ChalkObj): bool =
+  ## Inserts the chalk binary into a zip archive
+  ## Returns true if successful, false otherwise
+
+  result = false
+
+  # Get the currently executing chalk binary path
+  let myAppPath = getMyAppPath()
+
+  let chalkBinaryContent = tryToLoadFile(myAppPath)
+
+  let zipCache = ZipCache(chalk.cache)
+  # Get the paths to add the binary
+  let
+    contentDir = joinPath(zipCache.tmpDir, "contents")
+    contentTargetPath = joinPath(contentDir, "chalk")
+
+  # Make sure the directory exists
+  if dirExists(contentDir):
+    # Write the chalk binary to content directory
+    let contentSuccess = tryToWriteFile(contentTargetPath, chalkBinaryContent)
+
+    if contentSuccess:
+      # Make executable
+      contentTargetPath.makeExecutable()
+      trace(chalk.name & ": added chalk binary to content zip dir")
+
+      result = true
+    else:
+      error(chalk.name & ": failed to add chalk binary to zip directory")
+  else:
+    error(chalk.name & ": zip content directory does not exist")
```

Comment:
```suggestion
    raise newException(ValueError, "zip content directory does not exist")
```

Reply:
Refactored both `if`s to raise if not true.